src/main/rx/nrf24.c

```diff
@@ -0,0 +1,122 @@
+/*
+ * This file is part of Cleanflight.
+ *
+ * Cleanflight is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Cleanflight is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Cleanflight.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <stdbool.h>
+#include <stdint.h>
+
+#include <platform.h>
+#include "build_config.h"
+
+#ifdef USE_RX_NRF24
+#include "drivers/rx_nrf24l01.h"
+#include "rx/rx.h"
+#include "rx/nrf24.h"
+#include "rx/nrf24_cx10.h"
+#include "rx/nrf24_syma.h"
+#include "rx/nrf24_v202.h"
+
+
+uint16_t nrf24RcData[MAX_SUPPORTED_RC_CHANNEL_COUNT];
+STATIC_UNIT_TESTED uint8_t nrf24Payload[NRF24L01_MAX_PAYLOAD_SIZE];
+STATIC_UNIT_TESTED uint8_t nrf24NewPacketAvailable; // set true when a new packet is received
+
+typedef void (*protocolInitPtr)(const rxConfig_t *rxConfig, rxRuntimeConfig_t *rxRuntimeConfig);
```

Comment:
And possibly even deference pointer in call `(*protocolInit)(...);` to make it explicit that function pointer is used